src/js/grapl-cdk/lib/swarm.ts

```diff
@@ -0,0 +1,175 @@
+         * flexible enough to allow interesting use-cases (e.g. spot
+         * instances).
+         *
+         * see this github issue comment for more info:
+         *
+         * https://github.com/docker/machine/issues/1655#issuecomment-409407523
+         *
+         * "DescribeSecurityGroups" -- required to check whether the
+         * --amazonec2-security-group actually exists
+         *
+         * "CreateSecurityGroup" -- not sure why this is required
+         *
+         * "DescribeSubnets" -- required to find the subnet
+         *
+         * "DescribeKeyPairs" -- to validate whether the keypair
+         * actually exists
+         *
+         * "CreateKeyPair" -- not sure why this is required
+         *
+         * these spot instance permissions apply if docker-machine is
+         * invoked with --amazonec2-request-spot-instance:
+         *
+         * "DescribeSpotInstances"
+         * "DescribeSpotInstanceRequests"
+         * "RequestSpotInstances"
+         * "CancelSpotInstanceRequests"
+         *
+         * "DescribeInstances" -- required to tell when an instance is
+         * ready, what its IP address is, etc
+         *
+         * "RunInstances" -- required to run an AWS instance if not
+         * --amazonec2-request-spot-instance
+         *
+         * "StartInstances" -- docker-machine start
+         *
+         * "StopInstances" -- docker-machine stop or docker-machine
+         * kill
+         *
+         * "RebootInstances" -- docker-machine restart
+         *
+         * "TerminateInstances" -- docker-machine rm
+         *
+         * "CreateTags" -- required to set the Name tag and anything
+         * that's passed via --amazonec2-tags
+         *
+         * "route53:ListHostedZonesByName" -- required to find the
+         * hosted zone ID
+         *
+         * "route53:ChangeResourceRecordSets" -- required to add A
+         * records to make DGraph discoverable.
+         */
+        const statement = new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: [
+                "ec2:DescribeSecurityGroups",
+                "ec2:CreateSecurityGroup",
+                "ec2:DescribeSubnets",
+                "ec2:DescribeKeyPairs",
+                "ec2:CreateKeyPair",
+                "ec2:DescribeSpotInstances",
+                "ec2:DescribeSpotInstanceRequests",
+                "ec2:RequestSpotInstances",
+                "ec2:CancelSpotInstanceRequests",
+                "ec2:DescribeInstances",
+                "ec2:RunInstances",
+                "ec2:StartInstances",
+                "ec2:StopInstances",
+                "ec2:RebootInstances",
+                "ec2:TerminateInstances",
+                "ec2:CreateTags",
+                "route53:ListHostedZonesByName",
+                "route53:ChangeResourceRecordSets"
+            ],
+            resources: ["*"]
+            // resources: [
```

Comment:
I couldn't make this work, not sure why. I guess we can remove `accountId` and `region` from `SwarmProps` since they're unused.

Reply:
Was just wondering why they were params and not from `cdk.Aws.ACCOUNT_ID` and `cdk.Aws.REGION`.